### PR TITLE
community: QuantizedEmbedders

### PR DESCRIPTION
**Pull request by mosheber**
**Description:** 
  - Adding a new Embedding class for using optimum-intel quantized embedders.  
  - Added a simple embedding example in the integrations docs, and also a full tutorial in the cookbook.
  - Added documentation for usage and installation in docs/docs/integrations/platforms/optimum_intel.mdx.
  **Dependencies:** 
  - https://github.com/huggingface/optimum-intel.git
  - https://github.com/intel/intel-extension-for-pytorch
  **Twitter handle:** TODO

### CODE REVIEW

**Review by danielfleischer — COMMENTED** [review]
Looks fine, but I don't see where the dependencies needed are installed? e.g. https://github.com/langchain-ai/langchain/blob/master/libs/community/pyproject.toml

File: libs/community/langchain_community/embeddings/optimum_intel.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
+from typing import Any, Dict, List, Optional
+
+import pandas as pd
+import torch
+from langchain_core.embeddings import Embeddings
+from langchain_core.pydantic_v1 import BaseModel, Extra
+from optimum.intel import IPEXModel
```
Comment (danielfleischer) [inline libs/community/langchain_community/embeddings/optimum_intel.py]: If optimum.intel is not installed you can't load this module. You need to protect this code with try catch and with comments on how to install the dependencies. Look at other files on how to use 3rd party libraries.

Comment (mosheber) [reply libs/community/langchain_community/embeddings/optimum_intel.py]: good point, I added the ipex loading into the load_model function, as done in the HuggingFaceBgeEmbeddings class.

**Comment by mosheber** [comment]
> https://github.com/langchain-ai/langchain/blob/master/libs/community/pyproject.toml

that is the problem with the installation process of optimum and ipex, its not just specifying packages, its an entire, more convoluted processes,. Thats why I link to both installation guides in the class itself and optimum_intel.mdx.

**Review by danielfleischer — APPROVED** [review]
Looks good!

**Pull request merged**